src/pages/home/report/ReportActionsView.js

```diff
@@ -133,6 +133,11 @@ class ReportActionsView extends React.Component {
         if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
             this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(nextProps.reportActions);
             this.mostRecentIOUReportActionID = ReportActionsUtils.getMostRecentIOUReportActionID(nextProps.reportActions);
+
+            if (this.newMarkerReportActionID !== '') {
+                this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(nextProps.report, this.sortedAndFilteredReportActions)});
+            }
+
```

Comment:
Updating state in shouldComponentUpdate is actually an anti-pattern, this is more intended to determine if the component should update at all. This should live in componentDidUpdate

Reply:
Where should I do it?

Reply:
Ah I see is that why you were running into the flickering? Another idea would be to check if when the action is deleted, if Report.subscribeToNewActionEvent is called - if so you could put it in there